File: packages/expo-updates/android/src/main/java/expo/modules/updates/selectionpolicy/ReaperSelectionPolicy.java
package expo.modules.updates.selectionpolicy;

import org.json.JSONObject;

import java.util.List;

import expo.modules.updates.db.entity.UpdateEntity;

/**
 * Given a list of updates, implementations of this class should choose which of those updates to
 * automatically delete from disk and which ones to keep.
 */
public interface ReaperSelectionPolicy {
  List<UpdateEntity> selectUpdatesToDelete(List<UpdateEntity> updates, UpdateEntity launchedUpdate, JSONObject filters);
}

File: packages/expo-updates/android/src/main/java/expo/modules/updates/selectionpolicy/ReaperSelectionPolicy.kt
package expo.modules.updates.selectionpolicy

import expo.modules.updates.db.entity.UpdateEntity
import org.json.JSONObject

/**
 * Given a list of updates, implementations of this class should choose which of those updates to
 * automatically delete from disk and which ones to keep.
 */
interface ReaperSelectionPolicy {
  fun selectUpdatesToDelete(
    updates: List<UpdateEntity>,
    launchedUpdate: UpdateEntity?,
    filters: JSONObject?
  ): List<UpdateEntity>
}
